addons/phone_validation/models/mail_thread_phone.py

```diff
@@ -45,8 +45,10 @@ class PhoneMixin(models.AbstractModel):
     phone_mobile_search = fields.Char("Phone/Mobile", store=False, search='_search_phone_mobile_search')
 
     def _search_phone_mobile_search(self, operator, value):
-
-        if len(value) <= 2:
+        count = sum(1 for char in value if char.isdigit())
+        if count == 0:
+            return [(0, '=', 1)]
+        if count <= 2:
```

Comment:
Rename "count" to "digits_count".
Now that I'm seeing the error message again, checking that "digits_count" is less or equal to 2 should be enough, no need to return the void domain after all.

That way we have a consistent behavior, if you enter too few digits to search on.

In other words:
```py
digits_count = len([char for char in value if char.isdigit()])
if digits_count <= 2:
   raise UserError(_('Please enter at least 3 digits when searching on phone / mobile.'))
```